packages/esm-patient-chart-app/src/deceased/deceased.resource.ts

```diff
@@ -0,0 +1,128 @@
+import { openmrsFetch, showToast, toOmrsIsoString, toDateObjectStrict, showNotification } from '@openmrs/esm-framework';
+import { Observable } from 'rxjs';
+import { openmrsObservableFetch } from '@openmrs/esm-api';
+import useSWR from 'swr';
+import dayjs from 'dayjs';
+import React, { useMemo } from 'react';
+
+interface PersonFetchResponse {
+  person: { uuid: string; dead: boolean; deathDate: Date; type: string };
+}
+
+interface CauseOfDeathFetchResponse {
+  uuid: string;
+  value: string;
+}
+
+interface ConcentAnswersResponse {
+  answers: Array<{
+    uuid: string;
+    name: string;
+    display: string;
+  }>;
+}
+
+export function usePatientDeceased(patientUuid: string) {
+  const person = usePersonfromPatient(patientUuid);
+  const causeOfDeath = useCauseOfDeathConcept();
+  const conceptAnswers = useConceptAnswers(causeOfDeath.value);
+  const result = useMemo(() => {
+    return {
+      conceptAnswers: conceptAnswers.conceptAnswers,
+      deathDate: person.personObject?.deathDate,
+      personUuid: person.personObject?.uuid,
+      isDead: person.personObject?.dead,
+      isLoading: person.isPersonLoading && causeOfDeath.isCauseOfDeathLoading && causeOfDeath.isCauseOfDeathLoading,
+      isValidating:
+        person.isPersonValidating && conceptAnswers.isConceptAnswerValidating && causeOfDeath.isCauseOfDeathValidating,
+    };
+  }, [
+    causeOfDeath.isCauseOfDeathLoading,
+    causeOfDeath.isCauseOfDeathValidating,
+    conceptAnswers,
+    person.isPersonLoading,
+    person.isPersonValidating,
+    person.personObject?.dead,
+    person.personObject?.deathDate,
+    person.personObject?.uuid,
+  ]);
+  return result;
+}
+
+export function useSetDeceased(
```

Comment:
This isn't really a React hook, so if we you just change the name to not start with `use`, e.g., `markPatientDeceased()` or something we shouldn't need to ignore the linting around React hooks here.

Reply:
thanks Ian